common/src/main/java/io/netty/util/HashedWheelTimer.java

```diff
@@ -560,12 +561,11 @@ private long waitForNextTick() {
                 //
                 // See https://github.com/netty/netty/issues/356
                 if (PlatformDependent.isWindows()) {
-                    sleepTimeMs = sleepTimeMs / 10 * 10;
+                    sleepTimeNanos = sleepTimeNanos / 10 * 10;
```

Comment:
@johnou nope... I guess we would need to use a min of 10ms sec here ?

Reply:
yep

Reply:
shouldn't it be something like this? `Math.round(value / factor) * factor;` or is it fine without the round?

Reply:
for reference https://github.com/netty/netty/issues/356#issuecomment-5866627

Reply:
There is a very neat workaround from the thread I mentioned below: https://gist.github.com/apangin/39677210c09d0c2abc82178ed1c93371

Not sure if we're willing to go extra mile here and support Windows better.

Reply:
@vkostyukov ROFL... oh my god. I guess we could do on a followup.

@apangin I wonder what the license of the code is ?

Reply:
@normanmaurer The code is in public domain. Though I would not recommend using it in common software like netty because of its undocumented JDK-specific nature.

The odd fact is that on Windows the resolution of `Thread.sleep` is better than of `LockSupport.park` without the above hack. So I would suggest to stick with `Thread.sleep` on Windows for millisecond resolution and go for `LockSupport.park` on other platforms.

Reply:
@apangin thanks for the info... I will use `Thread.sleep` on windows then.

Reply:
@apangin done